### DISCUSSION TITLE
How to seed with typescript ? [To load an ES module, set "type": "module"]

### DISCUSSION THREAD

**Discussion by Extarys** [prisma/prisma#14062] [Q&A]
I've been trying to use the built-in seed feature, changing my file extension, and different inline config, but I mostly always end up with the same message or a similar error.

I use Nuxt 3, so at the root I have a tsconfig file, here are the options set by Nuxt 3:
```
  "compilerOptions": {
    "jsx": "preserve",
    "target": "ESNext",
    "module": "ESNext",
    "moduleResolution": "Node",
    "skipLibCheck": true,
    "strict": false,
    "allowJs": true,
    "noEmit": true,
    "resolveJsonModule": true,
    "allowSyntheticDefaultImports": true,
    "types": [
      "node"
    ],
    "baseUrl": "..",
  }
```

`package.json`
```
"prisma": {
    "seed": "dotenv -e .env.dev ts-node --compiler-options {\"module\":\"CommonJS\"} prisma/seed.ts"
  },
```

Error:

```
Running seed command `dotenv -e .env.dev ts-node --compiler-options {"module":"CommonJS"} prisma/seed.ts` ...
(node:999017) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/web/nuxt-app/prisma/seed.ts:1
import { PrismaClient } from '@prisma/client';
```

`prisma/seed.ts`

```
import { PrismaClient } from '@prisma/client'
const prisma = new PrismaClient()
import { fieldEncryptionMiddleware } from 'prisma-field-encryption';
import { v4 as uuidv4 } from 'uuid';

import { resourceType } from "../services/config/enums"


async function main() {
    prisma.$use(fieldEncryptionMiddleware())
   [...]
}
```

I have no idea what type of options I need to set on the seed command for ts-node to run it successfully.  Thanks for any insight you might provide.

**Comment by austincrim** [comment]
Hey @Extarys 👋,

The `seed` command itself is passing in a `--compiler-options` flag which is overwriting the `module` option in your `tsconfig`. Try removing that flag and re-running the seed command.
```
"prisma": {
    "seed": "dotenv -e .env.dev ts-node prisma/seed.ts"
  },
```

**Reply by Extarys** [reply]
Thanks for your reply @austincrim ! I hoped it would work, but I get:
```
Running seed command `dotenv -e .env.dev ts-node prisma/seed.ts` ...
(node:1056054) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
nuxt-app/prisma/seed.ts:1
import { PrismaClient } from "@prisma/client";
^^^^^^

SyntaxError: Cannot use import statement outside a module
```

**Reply by austincrim** [reply]
Have you set `"type": "module"` in your `package.json`?

**Reply by Extarys** [reply]
:/ If I add this to `package.json` I get:
```
Running seed command `dotenv -e .env.dev ts-node prisma/seed.ts` ...
TypeError [ERR_UNKNOWN_FILE_EXTENSION]: Unknown file extension ".ts" for nuxt-app/prisma/seed.ts
```

I also tried to remove `node_modules` and reinstall all dependencies, `"ts-node": "^10.8.1"` and `"typescript": "^4.7.4"` is in the devDependencies too.

**Reply by pantharshit00** [reply]
Hello @Extarys, our apologies for the delay in getting back to you! Due to the large number of questions in our community channels, unfortunately this slipped through the cracks. It's very important for us to address your question, so let us know in a comment if this is still an open discussion. 

Can you try `--transpile-only` flag here? 

```
 "prisma": {
    "seed": "dotenv -e .env.dev ts-node --transpile-only prisma/seed.ts"
  }
```

**Reply by pantharshit00** [reply]
Hello @Extarys, Can you please confirm that this issue was addressed? I am going through past discussion right now and just want to make sure everything was addressed.

**Reply by Extarys** [reply]
Thanks for the reminder. I created a new project and installed dotenv-cli and can run the seed with `dotenv -e .env ts-node prisma/seed.ts` without any hassle. Thank you very much!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
